### PR TITLE
Printable Graph support for nested graphs + sugar

### PR DESCRIPTION
**Pull request by jamesr66a**
Example: 

```
node {
  output: "a"
  op_type: "Constant"
  attribute {
    name: "value"
    t {
      data_type: INT32
      int32_data: 3
      name: "three"
    }
    type: TENSOR
  }
}
node {
  output: "b"
  op_type: "Constant"
  attribute {
    name: "value"
    t {
      data_type: INT32
      int32_data: 6
      name: "three"
    }
    type: TENSOR
  }
}
node {
  output: "keepgoing"
  op_type: "Constant"
  attribute {
    name: "value"
    t {
      data_type: BOOL
      int32_data: 1
      name: "three"
    }
    type: TENSOR
  }
}
node {
  output: "max_trip_count"
  op_type: "Constant"
  attribute {
    name: "value"
    t {
      data_type: INT32
      int32_data: 10
      name: "three"
    }
    type: TENSOR
  }
}
node {
  input: "max_trip_count"
  input: "b"
  input: "keepgoing"
  op_type: "DoWhile"
  attribute {
    name: "body"
    g {
      node {
        input: "a"
        input: "b"
        output: "my_local"
        op_type: "Add"
      }
      node {
        input: "a"
        input: "b"
        output: "b_out"
        op_type: "Sub"
      }
      node {
        input: "my_local"
        input: "b"
        output: "keepgoing_out"
        op_type: "Greater"
      }
      node {
        input: "b"
        input: "b"
        output: "user_defined_vals"
        op_type: "Add"
      }
      name: "body-net"
      input {
        name: "b"
        type {
          tensor_type {
            elem_type: INT32
            shape {
            }
          }
        }
      }
      input {
        name: "keepgoing"
        type {
          tensor_type {
            elem_type: BOOL
            shape {
            }
          }
        }
      }
      output {
        name: "b_out"
        type {
          tensor_type {
            elem_type: INT32
            shape {
            }
          }
        }
      }
      output {
        name: "keepgoing_out"
        type {
          tensor_type {
            elem_type: BOOL
            shape {
            }
          }
        }
      }
      output {
        name: "user_defined_vals"
        type {
          tensor_type {
            elem_type: INT32
            shape {
            }
          }
        }
      }
    }
    type: GRAPH
  }
}
name: "predict-net"
```
Before:

```
graph predict-net {
  %a = Constant[value = <Tensor>]()
  %b = Constant[value = <Tensor>]()
  %keepgoing = Constant[value = <Tensor>]()
  %max_trip_count = Constant[value = <Tensor>]()
  DoWhile[body = <Unknown>](%max_trip_count, %b, %keepgoing)
  return
}
```

After: 

```
graph predict-net {
  %a = Constant[value = <Scalar Tensor [3]>]()
  %b = Constant[value = <Scalar Tensor [6]>]()
  %keepgoing = Constant[value = <Scalar Tensor [1]>]()
  %max_trip_count = Constant[value = <Scalar Tensor [10]>]()
  DoWhile[body = <graph body-net>](%max_trip_count, %b, %keepgoing)
  return
}

graph body-net (
  %b[INT32, scalar]
  %keepgoing[BOOL, scalar]
) {
  %my_local = Add(%a, %b)
  %b_out = Sub(%a, %b)
  %keepgoing_out = Greater(%my_local, %b)
  %user_defined_vals = Add(%b, %b)
  return %b_out, %keepgoing_out, %user_defined_vals
}
```

### CODE REVIEW

File: onnx/helper.py [hunk 1]
```diff
@@ -343,20 +370,32 @@ def printable_value_info(v):
     return s
 
 
-def printable_node(node, prefix=''):
+def printable_node(node, prefix='', subgraphs=False):
```
Comment (bddppq) [inline onnx/helper.py]: Is this just for backward compatibility or is there real cases that we don't want to print the subgraphs?

Comment (houseroad) [reply onnx/helper.py]: Do we even have any case which contain subgraph as attribute?

Comment (jamesr66a) [reply onnx/helper.py]: I basically just wanted to leave it up to the caller to handle the subgraphs. Notice that if the flag is true there's an extra return value and the caller can do whatever they want with it

Comment (bddppq) [reply onnx/helper.py]: I would say let's always print the subgraphs, and do not have different number of return values. Returning graph objects in a print function doesn't sound appropriate to me.

Comment (jamesr66a) [reply onnx/helper.py]: I think that makes sense in isolation, but in context this is used in a recursive descent algorithm. Printing the graph within the op doesn't look good, so what's happening now is we propagate all graphs up to the top-level (print_graph) and print them one after the other. This makes it look like a normal SSA organized into basic blocks

**Comment by ezyang** [comment]
One downside of putting the nested graphs out-of-line is that the lexical scoping relationship is no longer textually clear; furthermore, the use of `name` implies that the same graph may be referenceable multiple times from a single function, when in fact, the structural implementation of the proto does not demonstrate this.

**Comment by jamesr66a** [comment]
Going ahead and merging this, we can revisit the printout format later if people feel strongly about it

**Pull request merged**